lib/overcommit/hook/pre_commit/case_conflicts.rb

```diff
@@ -0,0 +1,19 @@
+module Overcommit::Hook::PreCommit
+  # Checks for files that would conflict in case-insensitive filesystems
+  # Adapted from https://github.com/pre-commit/pre-commit-hooks
+  class CaseConflicts < Base
+    def run
+      repo_files = Set.new(Overcommit::GitRepo.all_files)
```

Comment:
Perhaps by adding a method to `GitRepo` to list the files in a particular directory or list of directories. Something like

```ruby
# Returns the names of files in the given paths that are tracked by git.
#
# @return [Array<String>] list of absolute file paths
def list_files(*paths)
  `git ls-files #{paths.join(' ')}`.
    split(/\n/).
    map { |relative_file| File.expand_path(relative_file) }.
    reject { |file| File.directory?(file) } # Exclude submodule directories
end

# Returns the names of all files that are tracked by git.
#
# @return [Array<String>] list of absolute file paths
def all_files
  list_files
end
```